src/terrain/terrain_generation/biome.cpp

```diff
@@ -0,0 +1,95 @@
+#include "biome.hpp"
+
+#include "../../logging.hpp"
+#include "noise.hpp"
+
+#include <json/json.h>
+
+#include <quill/Quill.h>
+#include <sol/sol.hpp>
+
+#include <filesystem>
+#include <map>
+#include <string>
+#include <vector>
+
+namespace terrain {
+
+namespace terrain_generation {
+
+Biome::Biome(std::string biome_name) {
```

Comment:
This should be multiple functions

and not just the ctor

also why does this own the lua state?

Reply:
> This should be multiple functions

Not really, all it needs to do is get a vector of ints for the macro map. When we get to the point of using multiple lua states I will have a better idea of what these functions will be. Until then I don't want to split this into multiple functions.

> and not just the ctor

Yes, but this pr is about lua not biome. I want to add json reading to biome to define macor-tiles, but until then biome will just be this vector.

> also why does this own the lua state?

Mostly because I don't want this map function to be visible to other lua scrips that run in other parts of the code. My understanding is that the lua load script is equivalent to the import function in python. Variables and functions defined in that file will exist for the life of the lua state.

Reply:
ok will be refactored later